kratos/includes/model_part.h

```diff
@@ -807,6 +807,21 @@ class KRATOS_API(KRATOS_CORE) ModelPart : public DataValueContainer, public Flag
         return false;
     }
 
+    /** Returns if the Properties corresponding to it's identifier exists in any of the model parts */
+    bool RecursivelyHasProperties(IndexType PropertiesId, IndexType MeshIndex = 0) const
+    {
+        auto pprop_it = GetMesh(MeshIndex).Properties().find(PropertiesId);
+        if(pprop_it != GetMesh(MeshIndex).Properties().end()) { // Property does exist
+            return true;
+        } else {
+            if(IsSubModelPart()) {
+                return mpParentModelPart->HasProperties(PropertiesId, MeshIndex);
```

Comment:
why not like this?
```suggestion
                return mpParentModelPart->RecursivelyHasProperties(PropertiesId, MeshIndex);
```

Also please add tests for this